### DISCUSSION TITLE
Turn Text From DB in to V2 Message

### DISCUSSION THREAD

**Discussion by LLCJK** [nextgenhealthcare/connect#5412] [Q&A] [answered]
I have a channel (all datatypes are set to raw) that is connecting to a postgres db in javascript and grabbing what is supposed to be a v2 message from a coulmn with a datatype of text.

Through the DB conenction I set the v2 message I want as such:

    v2 = result.getString('original_message')

I am then trying to manipulate the v2 message but it is originally a string. The only way I know to turn it in to a v2 object is to make it xml then convert that xml to v2 like this:

    var xmlv2 = new XML(SerializerFactory.getHL7Serializer().toXML(v2))
    msg = SerializerFactory.getSerializer('HL7V2').fromXML(xmlv2);

But then doing:

    var countNotes = msg['NTE'].length()

I get a "Cannot call method "length" of undefined" error. What am I doing wrong here?

**Comment by pacmano1** [comment]
Is the message in the DB XML or plain text with pipes?

**Reply by LLCJK** [reply]
It is plain text with the pipes

**Reply by pacmano1** [reply]
```JS
$c('hl7',this_json.hl7);   // just grabbing the string with the pipes in it.
var serializationProperties = SerializerFactory.getDefaultSerializationProperties('HL7V2');
msg = SerializerFactory.getSerializer('HL7V2', serializationProperties, null).toXML($c('hl7'));
```

**Reply by LLCJK** [reply]
so in my case it would be this?

     var serializationProperties = SerializerFactory.getDefaultSerializationProperties('HL7V2')
     msg = SerializerFactory.getSerializer('HL7V2', serializationProperties, null).toXML(v2)

If so I am still getting the same length error

**Reply by pacmano1** [reply]
Upload (don't paste) a sample message you are processing, de-identified of course.  Seems like it is not serialzing correctly.

**Reply by LLCJK** [reply]
[TEST V2 MESSAGE.txt](https://github.com/nextgenhealthcare/connect/files/9530049/TEST.V2.MESSAGE.txt)

**Comment by LLCJK** [comment]
Did I over do it on the de-id part?

**Reply by pacmano1** [reply]
Priorities.   NFL opener and I am a Bills fan.

**Comment by LLCJK** [comment]
Haha fair enough. Best of luck tonight!

**Comment by tonygermano** [comment, marked as answer]
I'm not sure how your message is stored in the database, but your sample message in the [other thread](https://github.com/nextgenhealthcare/connect/discussions/5412#discussioncomment-3599345) is using `\r\n` at the end of each segment when the segment delimiter should be just `\r`. This could also be the result of a copy/paste on windows. You can try the code below, which is like checking the Convert Line Breaks box on the [hl7v2 data type properties](https://docs.nextgen.com/bundle/Mirth_User_Guide_41/page/connect/connect/topics/c_HL7_v2_x_Data_Type_connect_ug.html).

```js
var serializationProperties = SerializerFactory.getDefaultSerializationProperties('HL7V2')
serializationProperties.put('convertLineBreaks', true)
msg = SerializerFactory.getSerializer('HL7V2', serializationProperties, null).toXML(v2)
```
